File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/UpdatePortfolioData.java
package com.seagosoft.tws.data;

import com.ib.client.Contract;
import com.ib.client.Decimal;

public class UpdatePortfolioData {

    public Contract contract;
    public Decimal position;
    public double marketPrice;
    public double marketValue;
    public double averageCost;
    public double unrealizedPNL;
    public double realizedPNL;
    public String accountName;

    public UpdatePortfolioData(Contract contract,
                               Decimal position,
                               double marketPrice,
                               double marketValue,
                               double averageCost,
                               double unrealizedPNL,
                               double realizedPNL,
                               String accountName) {
        this.contract = contract;
        this.position = position;
        this.marketPrice = marketPrice;
        this.marketValue = marketValue;
        this.averageCost = averageCost;
        this.unrealizedPNL = unrealizedPNL;
        this.realizedPNL = realizedPNL;
        this.accountName = accountName;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/UpdatePortfolioData.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Contract
import com.ib.client.Decimal


class UpdatePortfolioData(
    contract: Contract,
    var position: Decimal,
    var marketPrice: Double,
    var marketValue: Double,
    var averageCost: Double,
    var unrealizedPNL: Double,
    var realizedPNL: Double,
    var accountName: String
) {
    var contract: Contract = contract
}